doc/source/code_ref/coordinates.rst

```diff
@@ -0,0 +1,196 @@
+   >>> c.lat
+   <Latitude -30.0 deg>
+   >>> c.lon
+   <Longitude180 70.0 deg>
+   >>> c.rad
+   <Distance 695508.0 km>
+
+
+Design of the Coordinates Module
+--------------------------------
+
+This module works by defining a collection of ``Frames``
+(`sunpy.coordinates.frames`), which exists on a transformation graph, where the
+transformations between the coordinate frames are then defined and registered
+with the transformation graph (`sunpy.coordinates.transformations`).
+
+Positions within these ``Frames`` are stored as a ``Representation`` of a
+coordinate, a representation is a description of a point in a Cartesian,
+spherical or cylindrical system (`sunpy.coordinates.representation`). A frame
+that contains a representation of one or many points is said to have been
+'realized'.
+
+For a more in depth look at the design and concepts of the Astropy coordinates
+system see :ref:`astropy-coordinates-overview`
+
+
+
+Frames and SkyCoord
+^^^^^^^^^^^^^^^^^^^
+
+The `~astropy.coordinates.SkyCoord` class is a high level wrapper around the
+`astropy.coordinates` package. It provides an easier way to create and transform
+coordinates, by using string representations for frames rather than the classes
+themselves and some other usability improvements, for more information see the
+`~astropy.coordinates.SkyCoord` documentation.
+
+The main advantage provided by `~astropy.coordinates.SkyCoord` is the support it
+provides for caching Frame attributes. Frame attributes are extra data specified
+with a frame, some examples in `sunpy.coordinates` are ``dateobs`` or ``L0`` and
+``B0`` for observer location. Only the frames where this data is meaningful have
+these attributes, i.e. only the Helioprojective frames have ``L0`` and ``B0``.
+However, when you transform into another frame and then back to a projective
+frame using `SkyCoord` it will remember the attributes previously provided, and
+repopulate the final frame with them. If you were to do transformations using
+the Frames alone this would not happen.
+
+The most important implication for this in `sunpy.coordinates` is the ``RSun``
+parameter in the projective frames. If you create a projective frame with a
+``RSun`` attribute, if you convert back to a projective frame it will be set
+correctly. It should also be noted that, if you create a Heliographic frame and
+then transform to a projective frame with an ``RSun`` attribute, it will not
+match the ``rad`` coordinate in the Heliographic frame. This is because you may
+mean to be describing a point above the defined 'surface' of the Sun.
+
+
+Coordinates and WCS
+-------------------
+
+The `sunpy.coordinates` package provides a mapping between FITS-WCS CTYPE
+convention and the coordinate frames as defined in `sunpy.coordinates`. This is
+provided in the `sunpy.coordinates.wcs_utils` module and adds the mappings to
+the `astropy.wcs.utils.WCS_FRAME_MAPPINGS` list. This list is used by packages
```

Comment:
This is quite advanced for user docs, and is an implementation detail - maybe just say that ``wcs_to_celestial_frame`` from Astropy recognizes the frames from Sunpy when sunpy.coordinates is imported?